src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -149,19 +162,116 @@ public int getTotalMessageCount(){
   }
   
-  /* Returns all laptops */
-  public List<Laptop> getAllLaptops(){
-	 List<Laptop> laptops = new ArrayList<>();
-	 laptops.add(new Laptop("Dell","Black","Windows",15,800.00));
-	 laptops.add(new Laptop("Thinkpad","Black","Windows",13,1200.00));
-	 laptops.add(new Laptop("Dell","Grey","Windows",15,800.00));
-	 laptops.add(new Laptop("Dell","Black","Linux",15,800.00));
-	 laptops.add(new Laptop("Thinkpad","Black","Windows",15,1699.00));
-	 laptops.add(new Laptop("Macbook Pro","Black","Mac",13,1599.00));
-	 laptops.add(new Laptop("MacBook Pro","Black","Mac",15,1900.00));
-	 laptops.add(new Laptop("Macbook Pro","Grey","Mac",13,1599.00));
-	 laptops.add(new Laptop("MacBook Pro","Grey","Mac",15,1900.00));
-
-	 return laptops;
-  }
-}
\ No newline at end of file
+  /**
+   * @return list of all Laptops
+   */
+  public List<Laptop> getAlllaptopList(){
+	 return laptopList;
+  }
+  
+  /** Adds a laptop to the LaptopList
+   * @param Laptop to add to the list 
+   */
+  public void addLaptop(Laptop newLaptop) {
+	  laptopList.add(newLaptop);
+  }
+  
+  /**Returns a laptop list with laptops of a certain brand*/
+  public List<Laptop> searchBrand(String brand){
+	  //list of laptops that meet the brand requirements 
+	  List<Laptop> brandreturn=new ArrayList<>();
+	  Iterator<Laptop> it= laptopList.iterator(); 
+	  while(it.hasNext()) {
+		  if(it.next().getBrand()==brand.toLowerCase())
+			  brandreturn.add(it.next());
+	  }
+	  
+	  return brandreturn; 
+  }
+  
+  /**Returns a laptop list with laptops of a certain OS*/
+  public List<Laptop> searchOs(String os){
+	  //list of laptops that meet the brand requirements 
+	  List<Laptop> osreturn=new ArrayList<>();
+	  Iterator<Laptop> it= laptopList.iterator(); 	  
+	  while(it.hasNext()) {
+		  if(it.next().getOS()==os.toLowerCase())
+			  osreturn.add(it.next());
+	  }
+	  
+	  return osreturn; 
+  }
+  
+  /**Returns a laptop list with laptops of a certain color*/
+  public List<Laptop> searchColor(String color){
+	  //list of laptops that meet the brand requirements 
+	  List<Laptop> colorreturn=new ArrayList<>();
+	  Iterator<Laptop> it= laptopList.iterator(); 
+	  while(it.hasNext()) {
+		  if(it.next().getColor()==color.toLowerCase())
+			  colorreturn.add(it.next());
+	  }
+	  
+	  return colorreturn; 
+  }
+  
+  /**Returns a laptop list with laptops of a certain OS*/
+  public List<Laptop> searchOs(String os){
+	  //list of laptops that meet the brand requirements 
+	  List<Laptop> osreturn=new ArrayList<>();
+	  Iterator<Laptop> it= laptopList.iterator();  
+	  while(it.hasNext()) {
+		  if(it.next().getOS()==os.toLowerCase())
+			  osreturn.add(it.next());
+	  }
+	  
+	  return osreturn; 
+  }
+
+  /**Returns all laptops above that size, inclusive*/ 
+  public List<Laptop> searchSizeAbove(int size){
+	  List<Laptop> sizeAboveReturn=new ArrayList<>();
+	  Iterator<Laptop> it=laptopList.iterator();
+	  while(it.hasNext()) {
+		  if(it.next().getSize()>=size)
```

Comment:
I would always wrap the body of `if` statements inside `{ }` curly brackets.

https://google.github.io/styleguide/javaguide.html#s4.1.1-braces-always-used